parlai/utils/strings.py

```diff
@@ -42,6 +46,9 @@ def normalize_reply(text: str) -> str:
     new_text = new_text.strip()
     new_text = new_text.replace('  ', ' ')
 
+    if version > 1 and (new_text[-1] not in ['!', '.', '?', ')']):
```

Comment:
```suggestion
    if version > 1 and new_text and and new_text[-1] not in '!.?)"\''
```
Note the fix for empty string and the handling of quotations